tools/bpm_bench/bpm_bench.cpp

```diff
@@ -197,14 +202,18 @@ auto main(int argc, char **argv) -> int {
       while (!metrics.ShouldFinish()) {
         auto page_idx = dist(gen);
         auto *page = bpm->FetchPage(page_ids[page_idx], AccessType::Get);
+        if (page == nullptr) {
+          continue;
+        }
 
+        page->RLatch();
         char ch = page->GetData()[page_idx % 1024];
+        page->RUnlatch();
         if (ch == 0) {
           throw std::runtime_error("invalid data");
         }
 
         bpm->UnpinPage(page->GetPageId(), false, AccessType::Get);
-        page_idx += 1;
```

Comment:
we will need to keep this line in order to iterate all pages?

Reply:
No, as I mentioned in the commit message, the addition result is never used, as the variable is initialized at the beginning of the loop:
https://github.com/cmu-db/bustub/blob/71a0168dc490b3e361c592328843c64e74453819/tools/bpm_bench/bpm_bench.cpp#L198